src/transformers/pipelines.py

```diff
@@ -1664,6 +1665,270 @@ def __call__(
+        device (:obj:`int`, `optional`, defaults to :obj:`-1`):
+            Device ordinal for CPU/GPU supports. Setting this to -1 will leverage CPU, >=0 will run the model
+            on the associated CUDA device id.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        assert self.tokenizer.eos_token_id is not None, "DialoguePipeline tokenizer should have an EOS token set"
+        if self.tokenizer.pad_token_id is not None:
+            self.pad_token_id = self.tokenizer.pad_token_id
+        else:
+            self.pad_token_id = self.tokenizer.eos_token_id
+        self.min_response_allowed_length = kwargs.get("min_response_allowed_length", 32)
+
+    def __call__(self, *args, clean_up_tokenization_spaces=True, **generate_kwargs):
+        r"""
+        Args:
+            *args: (list of `:class:Conversation`) Conversations to generate responses for
+            **generate_kwargs: extra kwargs passed to `self.model.generate`_
+
+        Returns:
+            list of conversations with updated generated responses for those containing a new user input
+        """
+
+        active_conversations = []
+        # Input validation
+        if isinstance(args[0], list):
+            active_conversations_indices = dict()
+            active_index = 0
+            for conversation_index, conversation in enumerate(args[0]):
+                assert isinstance(
+                    conversation, Conversation
+                ), "DialoguePipeline expects a Conversation or list of Conversations as an input"
+                if conversation.new_user_input is None:
+                    logger.warning(
+                        "Conversation with id {} does not contain new user input and will not be updated".format(
+                            conversation.uuid
+                        )
+                    )
+                else:
+                    active_conversations_indices[conversation_index] = active_index
+                    active_conversations.append(conversation)
+                    active_index += 1
+            assert (
+                self.tokenizer.pad_token_id is not None or self.tokenizer.eos_token_id is not None
+            ), "Please make sure that the tokenizer has a pad_token_id or eos_token_id when using a batch input"
+        elif isinstance(args[0], Conversation):
+            active_conversations.append(args[0])
+        else:
+            raise ValueError("DialoguePipeline expects a Conversation or list of Conversations as an input")
+        if len(active_conversations) > 0:
+            with self.device_placement():
+
+                inputs = self._parse_and_tokenize(
+                    [conversation.new_user_input for conversation in active_conversations]
+                )
+                histories = [conversation.history for conversation in active_conversations]
+                max_length = generate_kwargs.get("max_length", 1000)
```

Comment:
I think I would prefer to not set the default `max_length` to 1000 here. The user can set the default value for each model individually in the model's config (under task specific params) => compare for example with XLNet here: https://s3.amazonaws.com/models.huggingface.co/bert/xlnet-base-cased-config.json .

Reply:
```suggestion
                max_length = generate_kwargs.get("max_length", self.model.config.max_length)
```

Reply:
I agree the value of 1000 is arbitrary (taken from the illustrative example in the model card). The issue is that the `DialoGPT` configuration files do not set a `max_length`. To my understanding, this means that without specifying it to the `generate` method, it will be set to the GPT2 default, that is 20. This seems very low for a conversation pipeline as the input alone is likely to exceed this value. I am not sure defaulting to the configuration value is going to be a good user experience. Maybe the way to go is to force the user to provide a value? Or maybe update the configuration of the `DialoGPT` configuration files?